include/network.h

```diff
@@ -107,6 +107,54 @@
       return str_sprintf("%02X:%02X:%02X:%02X:%02X:%02X", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
     }
 
-
-
+    class NetworkReader

```

Comment:
Maybe a comment, like what is NetworkReader and what does it do?   Just a summary would be handy!